src/components/Products/Products.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+import axios from 'axios';
+import Pagination from "./Pagination/Pagination";
+import Product from "./Product/Product";
+
+const Products = () => {
+  const [products, setProducts] = useState([]);
+  const [totalProducts, setTotalProducts] = useState();
+  const [currentPage, setCurrentPage] = useState(1);
+  const [productsPerPage, setProductsPerPage] = useState(16);
+
+  const pages = [];
+  for(let i = 1; i < Math.ceil(totalProducts / productsPerPage); i++){
+    pages.push(i);
+  }
```

Comment:
Không làm theo cách thủ công như này anh nhé. Chỉ cần
`const totalPage = Math.ceil(totalProducts / productsPerPage)`
Rồi truyền cho pagination dưới dạng props.
Rồi muốn in ra mảng thì chỉ cần
`[...Array(totalPage)].map`